cli/tsDocGen.ts

```diff
@@ -0,0 +1,207 @@
+  } catch (_err) {
+    // eslint-disable-next-line no-console
+    // console.warn(`No component found in ${filename}:`, err);
+  }
+
+  return (parsedComponents || []).filter(
+    (parsed) => parsed && parsed.displayName,
+  )
+}
+
+const getNodeText = (node, sourceText) => {
+  if (!node || !node.pos || !node.end) {
+    return undefined
+  }
+
+  return sourceText.substring(node.pos, node.end).trim()
+}
+
+const buildJsDocProps = (nodes, sourceText) =>
+  nodes?.reduce((acc, member) => {
+    const name =
+      (member.name && member.name.escapedText) ||
+      (member.parameters &&
+        `[${getNodeText(member.parameters[0], sourceText)}]`) ||
+      'Unknown'
+    acc[name] = {
+      description: member.jsDoc
+        ? member.jsDoc.map((doc) => doc.comment).join('\n')
+        : null,
+      required: member.questionToken === undefined,
+      type: {
+        raw: getNodeText(member.type, sourceText).trim(),
+      },
+    }
+    return acc
+  }, {})
+
+const getSourceFileStatements = (filename, sourceText) => {
+  const { statements } = ts.createSourceFile(
+    filename,
+    sourceText,
+    ts.ScriptTarget.Latest, // languageVersion
+  )
+
+  return statements
+}
+
+const getInterfaceMetadata = (filename, sourceText) =>
+  getSourceFileStatements(filename, sourceText).reduce(
+    (metaDataAcc, statement) => {
+      if (statement.kind === ts.SyntaxKind.InterfaceDeclaration) {
+        const _statement = statement as any
+        metaDataAcc.push({
+          displayName: _statement.name.escapedText,
+          description: _statement.jsDoc?.map((doc) => doc.comment).join('\n'),
+          props: buildJsDocProps(_statement.members, sourceText),
+        })
+      }
+
+      return metaDataAcc
+    },
+    [],
+  )
+
+const getTypeAliasMetadata = (filename, sourceText) =>
+  getSourceFileStatements(filename, sourceText).reduce(
+    (metaDataAcc, statement) => {
+      if (statement.kind === ts.SyntaxKind.TypeAliasDeclaration) {
+        const _statement = statement as any
+        const props = _statement.type.types?.reduce((propAcc, type) => {
+          if (type.members) {
+            propAcc.push(buildJsDocProps(type.members, sourceText))
+          }
+
+          return propAcc
+        }, [])
+
+        metaDataAcc.push({
+          props,
+          displayName: _statement.name.escapedText,
+          description: _statement.jsDoc?.map((doc) => doc.comment).join('\n'),
+        })
+      }
+
+      return metaDataAcc
+    },
+    [],
+  )
+
+interface Res {
+  name: string
+  type: string
+  description?: string
+  required?: boolean
+  defaultValue?: string
+  hide?: boolean
+}
+
+function normalizeProp([
+  name,
+  { required, annotatedType, type, tsType, description, defaultValue },
+]) {
+  const res: Res = {
+    name,
+    type:
+      annotatedType ||
+      (type && type.name) ||
+      (type && (type.raw || type.name)) ||
+      (tsType && (tsType.raw || tsType.name)) ||
+      'No type info',
```

Comment:
Could either remove this or update the PropsTable since I had included a fallback there with a similar string.

Reply:
Yeah we can if you want, I just wasn't trying to mess with tsDocGen any more than I needed to at the moment.